Python-Part4.ipynb

```diff
@@ -815,171 +939,279 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Pandas\n",
+    "## Pandas\n",
+    "\n",
+    "In this section, we will learn about a python package called `pandas` that contains very helpful functions and data structures for flat data types like the tab or comma-delimited files you might normally read into Excel. In previous iterations of this class, we taught both `pandas` and `numpy`, which is another library that `pandas` uses under the hood. In this workshop, we will focus on `pandas` only, and do a deeper dive. \n",
     "\n",
-    "In this section, we will learn about a python package called `pandas` that contains very helpful functions and data structures just for flat data types like the tab or comma-delimited files you might normally read in Excel. In previous iterations of this class, we taught both `pandas` and `numpy`. In this workshop, we will focus on `pandas` only, and do a deeper dive. "
+    "First, we'll learn about the basic data structures available in `pandas`, what their closest corollaries are in base Python, and what advantages they have over those corollaries."
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Importing data to pandas\n",
-    "One of the most useful features of pandas DataFrames is its ability to easily perform complex data transformations. This makes it a powerful tool for cleaning, filtering, and summarizing tabular data. As shown above, we can manually create a DataFrame from scratch, but more commonly you will want to read in data from an external source, such as the output of a bioinformatic program, and do some manipulation of it. Let's read some data into a DataFrame to demonstrate. \n",
     "\n",
-    "Below you can see an example of how to read files into pandas using the `pd.read_csv()` function. The `csv` stands for 'comma-separated values', which means by defaults it will assume that our columns are separated by **commas**; if we wanted to change the delimiter (e.g. in the case of a tab-separated file), we can set the delimiter explicitly using the `sep=` argument. "
+    "### `pandas` series\n",
+    "\n",
+    "A `Series` is the simplest data structure in `pandas`. They are one dimensional (1D) objects composed of a **single data type** of any variety (string, integers); you can basically think of them as a single column in a spreadsheet. In that sense they are similar to lists in base Python (and arrays in `numpy`), however unlike those other 1D structures Series also have **label-based indexing**, meaning each element in the object can be accessed by specifying its specific label. In that way, they are also similar to dictionaries in base Python. \n"
    ]
```

Comment:
Anything else we should include? Do they allow multiple data types? Are they mutable?